### PR TITLE
Feature state management

### PR DESCRIPTION
**Pull request by yanevdw**
Used Redux and Redux Toolkit to incorporate RTK Query to perform API calls.

This PR includes:
- Adding a Redux store
- Adding interfaces for the API responses
- Added an API service (`api.ts`) to query the API.

### CODE REVIEW

File: src/models/state.d.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  slug: string;
+  title: string;
+  demographic: string;
+  content_rating: string;
+  genres: number[];
+  last_chapter: number;
+  md_covers: MdCovers;
+}
+
+export interface MdCovers {
```
Comment (LesediMoko) [inline src/models/state.d.ts]: The type of ` vol ` is sometimes null in the API response (check the recommendation array of ` https://api.comick.io/comic/{comic_name}/ ` ). So consider making the type string | null.

File: src/models/state.d.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+export interface Relates {
+  title: string;
+  slug: string;
+  hid: string;
+  md_covers: MdCovers;
+}
+
+export interface Recommendations {
```
Comment (LesediMoko) [inline src/models/state.d.ts]: Consider making the interface names singular and make the properties of that type plural (if they are arrays)

i.e.
` export interface Recommendation {...} ` then ` recommendations : Recommendation[] `

File: src/models/state.d.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  title: string;
+}
+
+export interface MdRelates {
+  name: string;
+}
+
+export interface RelateFrom {
+  relate_to: RelateTo;
+  md_realtes: MdRelates;
```
Comment (LesediMoko) [inline src/models/state.d.ts]: oops, typo on line 67 😅

File: src/models/state.d.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+export interface Comic {
+  id: number;
+  hid: string;
+  title: string;
+  country: string;
+  links: Links;
+  last_chapter: number;
+  chapter_count: number;
+  dempgraphic: number;
```
Comment (LesediMoko) [inline src/models/state.d.ts]: *demographic 😅

**Review by LesediMoko — DISMISSED** [review]
Just a few comments regarding the interface naming, a couple of typos and also potential type mismatches to look out for. Everything else looks good to me!

**Review by LesediMoko — APPROVED** [review]
Changes I mentioned have been made. Nice!

**Review by Julianna-Venter — APPROVED** [review]
No comment from me

**Pull request merged**